### ISSUE TITLE
UI consistency for progress bar in TechDocs to Storybook components for waiting feedback

### ISSUE THREAD

**Issue by adamdmharvey** [backstage/backstage#7054]
## Feature Suggestion

It occurred to me that when visiting a docs page, the TechDocs frontend plugin shows the user a circular spinner.

<img width="851" alt="image" src="https://user-images.githubusercontent.com/33203301/131939826-05f0d836-24f8-452b-a040-a017917a62e2.png">

However, everywhere else throughout the site uses the progress bar spinner to suggest when something is loading as identified in the Storybook here:

<img width="879" alt="image" src="https://user-images.githubusercontent.com/33203301/131940095-3ee88aba-6dd0-45cf-a895-5c42e20056c4.png">

https://backstage.io/storybook/?path=/story/feedback-progress--progress

I wonder if the UI here should be better aligned to keep the styling consistent? It's obviously not a one-for-one trade for a circular spinner here, which sticks in the left of this box and is replaced by a green check when done, over to a horizontal progress bar, but some UI consistency might drive up formality/cleanliness of the UI.

## Possible Implementation

1. Add a circular spinner option to Storybook and ensure the design matches the overall Backstage design aesthetic
2. Modify these refreshing boxes in TechDocs router to potentially use the progress spinner at the bottom of the box so they sort of cleanly melt into the visual display of the box.

## Context

As referenced here:

https://github.com/backstage/backstage/blob/f38887d59c9d8af4ea4cb306e768775b0ffaee7e/plugins/techdocs/src/reader/components/Reader.tsx#L382-L390

and

https://github.com/backstage/backstage/blob/f38887d59c9d8af4ea4cb306e768775b0ffaee7e/plugins/techdocs/src/reader/components/Reader.tsx#L393-L402

PS: In testing this a bit more, I happened to notice the Cost Insights also uses a different overlay spinner that I hadn't seen before (at least I think it does).

Thought I'd just open this to see if there's any seconds to this, or if the different feedback seems OK to others in which case thats ok too!

**Comment by Rugvip** [comment]
Yep, definitely use our core `Progress` component! It was a big battle to get rid of all of the various progress components people were using internally in order to unify the design. Should not let that slip! x)

**Comment by iamEAP** [comment]
Just a note after trying out option 2 above...

We played around with this in #7476, but after doing so, we realized that maybe the `<Progress />` component is not appropriate in this case.  It really only makes sense to render during _parts_ of the TechDocs content refresh process (e.g. during generation), but eventually the reader lands in a state where docs _have been_ generated, and we're just waiting for the user to click `Refresh`.

It doesn't make sense to show the indicator at that point, but that means it's appearing/disappearing in the top of the page during the generation process, which causes the docs content to jump up and down quite a bit more than necessary.  Your reading would be disrupted if you didn't actually care about the page state and had gone further down to read the docs you were there to read in the first place.

To alleviate this jumping, I tried setting it to `definite` with progress `100` in that state, but the transition from `indefinite` to `definite` is awkward and the unchanging blue bar looks out of place once it's in that final state.

So I'd recommend other options.
